Firestore/core/test/firebase/firestore/util/hard_assert_test.cc

```diff
@@ -48,6 +48,21 @@ TEST(HardAssertTest, WithMessage) {
 }
 
 TEST(HardAssertTest, NonDefaultFailureHandler) {
+  // Used to ensure the original failure handler is restored.
+  class FailureHandlerRestorer {
```

Comment:
I like this idea. It's probably more broadly applicable than just this one test but rolling this out to other tests is something we can do later.